### PR TITLE
Turn kbaseNarrativeJobsPanel -> jobCommChannel

### PR DESCRIPTION
**Pull request by briehl**
That included a bunch of visual code that's not used, and doesn't really need to be a KBWidget to do its job. This rewrites that module into a fairly simple class that did things.

Marking this as a draft because there's still some outstanding issues, and I'm not sure if they're related. Namely, there's some messages that are being missed.

ETA: apparently I hit the wrong button, and can't seem to turn this as a draft. So, just don't merge yet!

ETA2: ready to merge. @bio-boris okayed it, so I'll merge once tests pass again.

### CODE REVIEW

**Comment by briehl** [comment]
@bio-boris This is probably ready enough for review. There's some more changes that build on it that are upcoming, but this is enough to gnaw on for a while.

If you have any questions, let me know!

File: kbase-extension/static/narrative_paths.js [hunk 1]
```diff
@@ -122,7 +122,7 @@ require.config({
         'kbaseNarrativeDataPanel': 'kbase/js/widgets/narrative_core/kbaseNarrativeDataPanel',
         'kbaseNarrativeDataList': 'kbase/js/widgets/narrative_core/kbaseNarrativeDataList',
         'kbaseNarrativeSidePanel': 'kbase/js/widgets/narrative_core/kbaseNarrativeSidePanel',
-        'kbaseNarrativeJobsPanel': 'kbase/js/widgets/narrative_core/kbaseNarrativeJobsPanel',
+        'jobCommChannel': 'kbase/js/widgets/narrative_core/jobCommChannel',
```
Comment (bio-boris) [inline kbase-extension/static/narrative_paths.js]: kbaseNarrativeJobsPanel is replaced with a jobCommChannel! Release notes worthy!

File: test/unit/spec/narrative_core/kbaseNarrativeJobsPanel-spec.js [hunk 2]
```diff
@@ -4,9 +4,9 @@
 /*global beforeEach, afterEach*/
 /*jslint white: true*/
 define([
-    'kbaseNarrativeJobsPanel'
+    'jobCommChannel'
 ], function(Widget) {
-    describe('Test the kbaseNarrativeJobsPanel widget', function() {
+    describe('Test the jobCommChannel widget', function() {
```
Comment (bio-boris) [inline test/unit/spec/narrative_core/kbaseNarrativeJobsPanel-spec.js]: I've never actually ran the tests on my local environment, I've just looked at travis believe.

Comment (briehl) [reply test/unit/spec/narrative_core/kbaseNarrativeJobsPanel-spec.js]: Huh, I guess I haven't updated those tests as part of this PR. I can put in a few.

File: kbase-extension/static/kbase/js/kbaseNarrative.js [hunk 3]
```diff
@@ -778,24 +780,20 @@ define([
...
-            $([Jupyter.events]).trigger('loaded.Narrative');
             $([Jupyter.events]).on('kernel_ready.Kernel',
-                function () {
-                    console.log('Kernel Ready! Initializing Job Channel...');
+                (e) => {
                     this.loadingWidget.updateProgress('kernel', true);
+                    this.jobCommChannel = new JobCommChannel();
                     // TODO: This should be an event "kernel-ready", perhaps broadcast
                     // on the default bus channel.
-                    this.sidePanel.$jobsWidget.initCommChannel()
```
Comment (bio-boris) [inline kbase-extension/static/kbase/js/kbaseNarrative.js]: This widget is removed... Was the widget even viewable before?

Comment (briehl) [reply kbase-extension/static/kbase/js/kbaseNarrative.js]: Nope. Once upon a time, there was a separate "Jobs" tab in the Narrative interface. That widget drove it. Then that got removed, and it wasn't visible anymore, but the widget was still needed for kernel communication.

Most of this PR is removing all of that old mess and turning the "Job Panel" widget just into a communication channel.

File: kbase-extension/static/narrative_paths.js (hunk 1 above)
Comment (briehl) [reply kbase-extension/static/narrative_paths.js]: Agreed!

File: test/unit/spec/narrative_core/kbaseNarrativeJobsPanel-spec.js (hunk 2 above)
Comment (briehl) [reply test/unit/spec/narrative_core/kbaseNarrativeJobsPanel-spec.js]: You can run tests with either `make test` to do the whole suite, or `make test-backend` or `make test-frontend-unit` to do either/or.

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeSidePanel.js [hunk 4]
```diff
@@ -1,410 +1,381 @@
 /*global define*/
```
Comment (bio-boris) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeSidePanel.js]: A lot of un-used or un-needed stuff is removed here.

**Comment by briehl** [comment]
Great. I'm working up some tests now. It's tricky with all of the back-and-forth among different components that need to be mocked (since it's a unit test), but I'l get there...

**Pull request merged**